src/assignment/views/AssignmentEdit.tsx

```diff
@@ -135,6 +135,12 @@ const AssignmentEdit: FunctionComponent<AssignmentEditProps> = ({
 		updateBlocksInAssignmentState
 	);
 
+	useEffect(() => {
+		setAssigmentLoading(true);
+		updateAssignmentEditor();
+		setAssigmentLoading(false);
+	}, []);
```

Comment:
if you don't wait for the updateAssignmentEditor then there is no point in setting the loading state, since it will happen instantly
you can await inside a useEffect, so my solution is:
```suggestion
	const updateAssignmentEditorWithLoading = useCallback(async () => {
		setAssigmentLoading(true);
		await updateAssignmentEditor();
		setAssigmentLoading(false);
	}, [setAssigmentLoading]);
	
	useEffect(() => {
		updateAssignmentEditorWithLoading();
	}, [updateAssignmentEditorWithLoading]);
```

or maybe set the loading state inside the updateAssignmentEditor function?